Warning in lld/COFF/Writer.cpp while building llvm

```
│ │ $SRC_DIR/lld/COFF/Writer.cpp:1740:27: warning: comparison of integers of different signs: 'uint32_t' (aka 'unsigned int') and 'long' [-Wsign-compare]
 │ │  1740 |   assert(coffHeaderOffset == buf - buffer->getBufferStart());
 │ │       |          ~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 │ │ $BUILD_PREFIX/opt/emsdk/upstream/emscripten/cache/sysroot/include/assert.h:8:28: note: expanded from macro 'assert'
 │ │     8 | #define assert(x) ((void)((x) || (__assert_fail(#x, __FILE__, __LINE__, __func__),0)))
 │ │       |                            ^
 │ │ $SRC_DIR/lld/COFF/Writer.cpp:788:7: note: in instantiation of function template specialization '(anonymous namespace)::Writer::writeHeader<llvm::object::pe32plus_header>' requested here
 │ │   788 |       writeHeader<pe32plus_header>();
 │ │       |       ^
 │ │ $SRC_DIR/lld/COFF/Writer.cpp:1773:25: warning: comparison of integers of different signs: 'uint32_t' (aka 'unsigned int') and 'long' [-Wsign-compare]
 │ │  1773 |   assert(peHeaderOffset == buf - buffer->getBufferStart());
 │ │       |          ~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 │ │ $BUILD_PREFIX/opt/emsdk/upstream/emscripten/cache/sysroot/include/assert.h:8:28: note: expanded from macro 'assert'
 │ │     8 | #define assert(x) ((void)((x) || (__assert_fail(#x, __FILE__, __LINE__, __func__),0)))
 │ │       |                            ^
 │ │ $SRC_DIR/lld/COFF/Writer.cpp:1840:26: warning: comparison of integers of different signs: 'uint32_t' (aka 'unsigned int') and 'long' [-Wsign-compare]
 │ │  1840 |          dataDirOffset64 == buf - buffer->getBufferStart());
 │ │       |          ~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 │ │ $BUILD_PREFIX/opt/emsdk/upstream/emscripten/cache/sysroot/include/assert.h:8:28: note: expanded from macro 'assert'
 │ │     8 | #define assert(x) ((void)((x) || (__assert_fail(#x, __FILE__, __LINE__, __func__),0)))
 │ │       |                            ^
 │ │ $SRC_DIR/lld/COFF/Writer.cpp:1740:27: warning: comparison of integers of different signs: 'uint32_t' (aka 'unsigned int') and 'long' [-Wsign-compare]
 │ │  1740 |   assert(coffHeaderOffset == buf - buffer->getBufferStart());
 │ │       |          ~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 │ │ $BUILD_PREFIX/opt/emsdk/upstream/emscripten/cache/sysroot/include/assert.h:8:28: note: expanded from macro 'assert'
 │ │     8 | #define assert(x) ((void)((x) || (__assert_fail(#x, __FILE__, __LINE__, __func__),0)))
 │ │       |                            ^
 │ │ $SRC_DIR/lld/COFF/Writer.cpp:790:7: note: in instantiation of function template specialization '(anonymous namespace)::Writer::writeHeader<llvm::object::pe32_header>' requested here
 │ │   790 |       writeHeader<pe32_header>();
 │ │       |       ^
 │ │ $SRC_DIR/lld/COFF/Writer.cpp:1773:25: warning: comparison of integers of different signs: 'uint32_t' (aka 'unsigned int') and 'long' [-Wsign-compare]
 │ │  1773 |   assert(peHeaderOffset == buf - buffer->getBufferStart());
 │ │       |          ~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 │ │ $BUILD_PREFIX/opt/emsdk/upstream/emscripten/cache/sysroot/include/assert.h:8:28: note: expanded from macro 'assert'
 │ │     8 | #define assert(x) ((void)((x) || (__assert_fail(#x, __FILE__, __LINE__, __func__),0)))
 │ │       |                            ^
 │ │ $SRC_DIR/lld/COFF/Writer.cpp:1840:26: warning: comparison of integers of different signs: 'uint32_t' (aka 'unsigned int') and 'long' [-Wsign-compare]
 │ │  1840 |          dataDirOffset64 == buf - buffer->getBufferStart());
 │ │       |          ~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 │ │ $BUILD_PREFIX/opt/emsdk/upstream/emscripten/cache/sysroot/include/assert.h:8:28: note: expanded from macro 'assert'
 │ │     8 | #define assert(x) ((void)((x) || (__assert_fail(#x, __FILE__, __LINE__, __func__),0)))
```

--- a/lld/COFF/Writer.cpp
+++ b/lld/COFF/Writer.cpp
@@ -1856,7 +1856,8 @@ template <typename PEHeaderTy> void Writer::writeHeader() {
   buf += sizeof(PEMagic);
 
   // Write COFF header
-  assert(coffHeaderOffset == buf - buffer->getBufferStart());
+  assert(coffHeaderOffset ==
+         static_cast<size_t>(buf - buffer->getBufferStart()));
   auto *coff = reinterpret_cast<coff_file_header *>(buf);
   buf += sizeof(*coff);
   SymbolTable &symtab =
@@ -1882,7 +1883,7 @@ template <typename PEHeaderTy> void Writer::writeHeader() {
       sizeof(PEHeaderTy) + sizeof(data_directory) * numberOfDataDirectory;
 
   // Write PE header
-  assert(peHeaderOffset == buf - buffer->getBufferStart());
+  assert(peHeaderOffset == static_cast<size_t>(buf - buffer->getBufferStart()));
   auto *pe = reinterpret_cast<PEHeaderTy *>(buf);
   buf += sizeof(*pe);
   pe->Magic = config->is64() ? PE32Header::PE32_PLUS : PE32Header::PE32;
@@ -1949,7 +1950,8 @@ template <typename PEHeaderTy> void Writer::writeHeader() {
 
   // Write data directory
   assert(!ctx.config.is64() ||
-         dataDirOffset64 == buf - buffer->getBufferStart());
+         dataDirOffset64 ==
+             static_cast<size_t>(buf - buffer->getBufferStart()));
   auto *dir = reinterpret_cast<data_directory *>(buf);
   buf += sizeof(*dir) * numberOfDataDirectory;
   if (symtab.edataStart) {

